flink-core/src/main/java/org/apache/flink/core/io/SimpleVersionedSerializerTypeSerializerProxy.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.core.io;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.typeutils.TypeSerializer;
+import org.apache.flink.api.common.typeutils.TypeSerializerSnapshot;
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataOutputView;
+import org.apache.flink.util.InstantiationUtil;
+import org.apache.flink.util.function.SerializableSupplier;
+
+import java.io.IOException;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * A {@link TypeSerializer} that delegates to an underlying {@link SimpleVersionedSerializer}.
+ *
+ * <p>This should not be used as a general {@link TypeSerializer}. It's meant to be used by internal
+ * operators that need to work with both {@link SimpleVersionedSerializer} and {@link
+ * TypeSerializer}.
+ */
+@Internal
+public class SimpleVersionedSerializerTypeSerializerProxy<T> extends TypeSerializer<T> {
+
+	private final SerializableSupplier<SimpleVersionedSerializer<T>> serializerSupplier;
+	private transient SimpleVersionedSerializer<T> cachedSerializer;
+
+	public SimpleVersionedSerializerTypeSerializerProxy(
+			SerializableSupplier<SimpleVersionedSerializer<T>> serializerSupplier) {
+		this.serializerSupplier = checkNotNull(serializerSupplier, "serializerSupplier");
+	}
+
+	@Override
+	public boolean isImmutableType() {
+		return false;
+	}
+
+	@Override
+	public TypeSerializer<T> duplicate() {
+		final byte[] serializedSerializer;
```

Comment:
Very sorry for the late comment. I think this method might be simplified by directly call InstantiationUtil.clone ?

Reply:
Not too late at all, thanks!